tests/exception/t13115.nim

```diff
@@ -1,12 +1,23 @@
-discard """
-  exitcode: 1
-  targets: "c"
-  matrix: "-d:debug; -d:release"
-  outputsub: ''' and works fine! [Exception]'''
-"""
+const msg = "This char is `" & '\0' & "` and works fine!"
 
-# bug #13115
-# xxx bug: doesn't yet work for cpp
-
-var msg = "This char is `" & '\0' & "` and works fine!"
-raise newException(Exception, msg)
+when defined nim_t13115:
+  # bug #13115
+  template fn =
+    raise newException(Exception, msg)
+  when defined nim_t13115_static:
+    static: fn()
+  fn()
+else:
+  import std/[osproc,strformat,os,strutils]
+  proc main =
+    const nim = getCurrentCompilerExe()
+    const file = currentSourcePath
+    for b in "c js cpp".split:
+      when defined(openbsd):
+        if b == "js": continue # xxx bug: pending #13115
+      for opt in ["-d:nim_t13115_static", ""]:
+        let cmd = fmt"{nim} r -b:{b} -d:nim_t13115 {opt} --hints:off {file}"
+        let (outp, exitCode) = execCmdEx(cmd)
+        doAssert msg in outp, cmd & "\n" & msg
+        doAssert exitCode == 1
+  main()
```

Comment:
Please get rid of this abomination.

Reply:
testament spec DSL isn't flexible enough for this test, sorry. If you disagree, please show a complete example that tests VM, openbsd (for non-js case), the '\0' being part of the output message, etc.

testament spec may be good for the general case, but you'll always have special case, which need things like trunner.nim, this test, tests/stdlib/tosproc.nim etc.